### PR TITLE
ENH added jinja2 variable spacing to lints

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -494,6 +495,25 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
+        with io.open(meta_fname, "rt") as fh:
+            for i, line in enumerate(fh.readlines()):
+                for m in JINJA_VAR_PAT.finditer(line):
+                    if m.group(1) is not None:
+                        var = m.group(1)
+                        if var != " %s " % var.strip():
+                            bad_vars.append(m.group(1).strip())
+                            bad_lines.append(i+1)
+        if bad_vars:
+            lints.append(
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:511]:
```suggestion
            hints.append(
```
?

Comment (beckermr) [reply conda_smithy/lint_recipe.py:511]: The spacing around jinja2 variables that we set are reported as lints and not hints:

```
    # 20: Jinja2 variable definitions should be nice.
    if recipe_dir is not None and os.path.exists(meta_fname):
        bad_jinja = []
        bad_lines = []
        # Good Jinja2 variable definitions look like "{% set .+ = .+ %}"
        good_jinja_pat = re.compile(r"\s*\{%\s(set)\s[^\s]+\s=\s[^\s]+\s%\}")
        with io.open(meta_fname, "rt") as fh:
            for jinja_line, line_number in jinja_lines(fh):
                if not good_jinja_pat.match(jinja_line):
                    bad_jinja.append(jinja_line)
                    bad_lines.append(line_number)
        if bad_jinja:
            lints.append(
                "Jinja2 variable definitions are suggested to "
                "take a ``{{%<one space>set<one space>"
                "<variable name><one space>=<one space>"
                "<expression><one space>%}}`` form. See lines "
                "{}".format(bad_lines)
            )
```

See line 384.

Why should this be different?

Comment (isuruf) [reply conda_smithy/lint_recipe.py:511]: IMO, they should start off as hints and then upgraded to lints. There was a few people complaining when lint 20 became a lint all of a sudden.

Comment (beckermr) [reply conda_smithy/lint_recipe.py:511]: hmmm ok. People complain a lot so this doesn't worry me, but ok.

Comment (isuruf) [reply conda_smithy/lint_recipe.py]: I was one of them. :smile:

Comment (beckermr) [reply conda_smithy/lint_recipe.py]: LLOL. Still not worried, but I respect your opinion! ;p

**Comment by beckermr** [comment]
cc @ocefpaf

**Pull request merged**